### PR TITLE
Make CollectionPage the default Schema.org type for archive queries

### LINKED ISSUE #4666: Missing @type in schema (ld+json) in archive templates
## Bug Description

The schema is malformed and missing the `@type` field in the archive templates (archive, tag, taxonomy, etc) therefore they have errors in the structured data

## Expected Behaviour

The schema for archive templates should be well formed with all its required fields. Namely, it should have a `@type` of `collection`.

## Steps to reproduce

1. Install the plugin and activate it
2. Register a custom taxonomy 
3. Create some posta and assign the taxonomy created to them
4. Go to the [validation tool](https://search.google.com/structured-data/testing-tool/u/0/)
5. Perform a test and see the error:  Unspecified Type (The @type is required and cannot be an empty string.)

## Additional context

 - WordPress version: 5.4.x
 - Plugin version: 1.4.2
 - Gutenberg plugin version (if applicable): NONE
 - AMP plugin template mode: 
 - PHP version: 7.1.x
 - OS: 
 - Browser: ALL
 - Device: ALL

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

Fixes #4666

## Checklist

- [x] My pull request is addressing an open issue (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by westonruter** [comment]
Question: When the queried object is a user, should the `@type` not be `collection` but something more “authory”?

**Comment by pierlon** [comment]
The Yoast SEO plugin identifies these pages as a [CollectionPage](https://github.com/Yoast/wordpress-seo/blob/19e029e95929c5f5c0cd4b3d9e93915febea3270/src/context/meta-tags-context.php#L386-L390), so perhaps it would be best to follow suit.

File: includes/amp-helper-functions.php [hunk 1]
```diff
@@ -1560,10 +1560,12 @@ function amp_get_schemaorg_metadata() {
...
 		 *
-		 * @param array   $metadata Metadata.
-		 * @param WP_Post $post     Post.
+		 * @param array   $metadata       Metadata.
+		 * @param WP_Post $queried_object Post.
 		 */
-		$metadata = apply_filters( 'amp_post_template_metadata', $metadata, $post );
+		$metadata = apply_filters( 'amp_post_template_metadata', $metadata, $queried_object );
+	} elseif ( is_archive() ) {
+		$metadata['@type'] = 'collection';
```
Comment (schlessera) [inline includes/amp-helper-functions.php]: I agree with @pierlon, we should use `CollectionPage` here, as it is a `WebPage` _containing_  a `Collection`.

Also, I'd stick to the casing of the official type reference.
```suggestion
		$metadata['@type'] = 'CollectionPage';
```

File: tests/php/test-amp-helper-functions.php [hunk 2]
```diff
@@ -1211,6 +1211,11 @@ public function test_amp_get_schemaorg_metadata() {
 		);
 		$this->assertEquals( $metadata['publisher']['logo'], amp_get_publisher_logo() );
 
+		// Test author archive.
+		$this->go_to( get_author_posts_url( $user_id ) );
+		$metadata = amp_get_schemaorg_metadata();
+		$this->assertEquals( 'collection', $metadata['@type'] );
```
Comment (schlessera) [inline tests/php/test-amp-helper-functions.php]:
```suggestion
		$this->assertEquals( 'CollectionPage', $metadata['@type'] );
```

**Review by schlessera — CHANGES_REQUESTED** [review]

**Comment by pierlon** [comment]
Seems like Travis lost its way back to this PR to give its report. Maybe another push would help?

**Review by pierlon — APPROVED** [review]

**Pull request merged**